jvb/src/main/java/org/jitsi/videobridge/octo/OctoEndpointMessageTransport.java

```diff
@@ -73,12 +73,27 @@ public void unhandledMessage(BridgeChannelMessage message)
     public BridgeChannelMessage addReceiver(@NotNull AddReceiverMessage message)
     {
         Conference conference = octoEndpoints.getConference();
-        AbstractEndpoint endpoint = conference.getEndpoint(message.getEndpointId());
-        // Since we currently broadcast everything in Octo, we may receive messages intended for another bridge. Handle
-        // only those that reference an endpoint local to this bridge.
-        if (endpoint instanceof Endpoint)
+
+        if (MultiStreamConfig.config.getEnabled())
+        {
+            String sourceName = message.getSourceName();
+            AbstractEndpoint endpoint = conference.findSourceOwner(sourceName);
```

Comment:
You could lookup the endpoint via `message.getEndpointId()` instead of source name (if you decide to keep using it)

Reply:
I removed the OctoV1 part and will try to only use the source name. Eventually the lookup by source name will have to be optimized anyway.